src/serlo-editor/plugins/text/hooks/use-editable-paste-handler.tsx

```diff
@@ -27,93 +26,41 @@ export const useEditablePasteHandler = (args: UseEditablePasteHandlerArgs) => {
 
   return useCallback(
     (event: React.ClipboardEvent) => {
-      const isListActive = isSelectionWithinList(editor)
-
-      const document = selectDocument(store.getState(), id)
-      if (!document) return
-
-      const mayManipulateSiblings = selectMayManipulateSiblings(
-        store.getState(),
-        id
-      )
-      if (!mayManipulateSiblings) return
+      // Exit if unable to select document data or if not allowed to manipulate siblings
+      const storeState = store.getState()
+      const document = selectDocument(storeState, id)
+      const mayManipulateSiblings = selectMayManipulateSiblings(storeState, id)
+      if (!document || !mayManipulateSiblings) return
 
+      // Exit if no files or text in clipboard data
       const files = Array.from(event.clipboardData.files)
       const text = event.clipboardData.getData('text')
-
-      // Handle pasted images or image URLs
-      if (files?.length > 0 || text) {
-        const imagePlugin = editorPlugins.getByType(EditorPluginType.Image)
-        if (!imagePlugin) return
-
-        const imagePluginState =
-          imagePlugin.onFiles?.(files) ?? imagePlugin.onText?.(text)
-
-        if (imagePluginState !== undefined) {
-          if (isListActive) {
-            showToastNotice(textStrings.noElementPasteInLists, 'warning')
-            return
-          }
-
-          insertPlugin({
-            pluginType: EditorPluginType.Image,
-            editor,
-            store,
-            id,
-            dispatch,
-            state: imagePluginState.state,
-          })
-          return
+      if (!files.length && !text) return
+
+      // Iterate through all plugins and try to process clipboard data
+      let media
+      for (const { plugin, type } of editorPlugins.getAllWithData()) {
+        const state = plugin.onFiles?.(files) ?? plugin.onText?.(text)
+        if (state?.state) {
+          media = { state: state.state as unknown, pluginType: type }
+          break
         }
       }
 
-      if (text) {
-        // Handle pasted video URLs
-        const videoPluginState = editorPlugins
-          .getByType(EditorPluginType.Video)
-          ?.onText?.(text)
-        if (videoPluginState !== undefined) {
-          event.preventDefault()
+      // Exit if no media was processed from clipboard data
+      if (media?.state === undefined) return
```

Comment:
`if (!media) return` should be enough here.

Reply:
I have a vague memory of trying that first, and then coming across some bug because of it. But definitely worth trying again, for brevity.